packages/pyright-internal/src/common/realFileSystem.ts

```diff
@@ -400,54 +400,12 @@ interface WorkspaceFileWatcher extends FileWatcher {
 export class WorkspaceFileWatcherProvider implements FileWatcherProvider {
     private _fileWatchers: WorkspaceFileWatcher[] = [];
 
-    constructor(private _workspaceMap: WorkspaceMap, private _console: ConsoleInterface) {}
-
-    createFileWatcher(paths: string[], listener: FileWatcherEventHandler): FileWatcher {
-        // Determine which paths are located within one or more workspaces.
-        // Those are already covered by existing file watchers handled by
-        // the client.
-        const workspacePaths: string[] = [];
-        const nonWorkspacePaths: string[] = [];
-
-        const workspaces = this._workspaceMap.getNonDefaultWorkspaces();
-        paths.forEach((path) => {
-            if (workspaces.some((workspace) => containsPath(workspace.rootPath, path))) {
-                workspacePaths.push(path);
-            } else {
-                nonWorkspacePaths.push(path);
-            }
-        });
-
-        // For any non-workspace paths, use the node file watcher.
-        const nodeWatchers = nonWorkspacePaths
-            .map((path) => {
-                // Skip paths that don't exist; fs.watch will throw when it tries to watch them,
-                // and won't give us a watcher that would work if it were created later.
-                if (!fs.existsSync(path)) {
-                    return undefined;
-                }
-
-                try {
-                    return fs.watch(path, { recursive: true }, (event, filename) =>
-                        listener(event as FileWatcherEventType, filename)
-                    );
-                } catch (e: any) {
-                    this._console.warn(`Exception received when installing file system watcher: ${e}`);
-                    return undefined;
-                }
-            })
-            .filter(isDefined);

```

Comment:
fyi @erictraut. I'm not familiar with non-vscode clients but removing this might affect them.

Reply:
VSCode added support for filewatching outside the workspace

Reply:
We were under the assumption that outside the workspace filewatching has also been broken.  But maybe it was only windows.

Reply:
unfortunately, it never worked before. 

basically, this code (https://github.com/microsoft/pyright/blob/main/packages/pyright-internal/src/analyzer/service.ts#L1414) ignored any events from nodejs since our code expects full path for the changed file (the behavior of vscode file watcher), but nodejs returns just filename. 

I confirm the behavior on both windows, and linux. pretty sure it will be same on mac.

basically, nodejs fs watcher never worked.

and on how to fix it, we decided to just rely on new vscode ability (https://code.visualstudio.com/updates/v1_64#_vscodeworkspacecreatefilesystemwatcher-now-supports-any-path) which will be supported by LSP 3.17 (which we are planning to move to https://github.com/microsoft/pyrx/pull/2289)

if we worry about non vscode client and keep old behavior on pyright and fix nodejs fs watcher issue rather than moving to vscode client (lsp 3.17) 's new fs ability, we can do that. but I dont think we need to that in this PR since outside of workspace nodejs file watcher was never worked before.  (at least last 2 years)